src/Adapter/ContextLoadHelper.php

```diff
@@ -0,0 +1,140 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter;
+
+use Context;
+use Currency;
+use DummyController;
+use Employee;
+use Shop;
+
+/**
+ * Helps loading specific context
+ */
+class ContextLoadHelper
+{
+    /**
+     * @var Context
+     */
+    private $context;
+
+    /**
+     * @param LegacyContext $context
+     */
+    public function __construct(LegacyContext $context)
+    {
+        $this->context = $context->getContext();
+    }
+
+    public function loadGenericContext(
```

Comment:
I'm wondering if `loadGenericContext()` should accept optional parameters. Maybe users will use each function separately if they want to use accurate identifiers ?

Reply:
So `loadGenericContext()` would have no parameter at all but simply call all the load methods with no parameters, which results in using the default fallback?

That's what I imagined at first indeed, some method you can call out of the box without needing to think of what happens behind